### PR TITLE
adding a GVCFWriter test against a sample GVCF

### LINKED ISSUE #736: port GVCFWriter
needed for HaplotypeCaller

### PR DESCRIPTION
**Pull request by lbergelson**
should fix #736

### CODE REVIEW

**Comment by lbergelson** [comment]
@akiezun Please review.  The test is hideous, but I'm not sure if there's a way to make it much cleaner.

File: src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java [hunk 1]
```diff
@@ -78,7 +78,13 @@ static int genotypeQualityFromPLs(final int[] minPLs){
             throw new GATKException("This should be a home ref block, but the lowest pl was not for hom ref");
         }
 
-        return sortedPls[1] - sortedPls[0];
+        final int rawQuality = sortedPls[1] - sortedPls[0];
+
+        // cap the quality to the highest quality that will be emitted by the VCFEncoder
+        // this isn't strictly necessary since it will be capped when written anyway
+        // it should help avoid confusion by preventing the quality from changing when written and loaded from disk
+        final int cappedQuality = Math.min(rawQuality, VCFConstants.MAX_GENOTYPE_QUAL);
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java]: inline the temp

File: src/test/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriterUnitTest.java [hunk 2]
```diff
@@ -478,4 +499,82 @@ private static void assertGVCFIsParseableAndVariantsMatch(File variantFile, List
         }
 
     }
+
+    @Test
+    public void testAgainstExampleGVCF() throws IOException {
+        final List<Integer> gqPartitions = Arrays.asList(1, 10, 20, 30, 40, 50, 60);
+        final File comparisonFile = new File("src/test/resources/org/broadinstitute/hellbender/utils/variant/writers/small.g.vcf");
+        final File outputFile =  new File("tmp.g.vcf");//createTempFile("generated", ".gvcf");
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriterUnitTest.java]: why not createTempFile?

File: src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java [hunk 3]
```diff
@@ -148,7 +149,8 @@ public void testIsContiguous(final String contig, final int pos, final boolean e
         return new Object[][] {
                 {new int[] {0, 1, 5}, 1},
                 {new int[] {10, 50, 30}, 20},
-                {new int[] {0, 10, 20, 10, 5, 50}, 5}
+                {new int[] {0, 10, 20, 10, 5, 50}, 5},
+                {new int[] {0, 1000, 1000000}, VCFConstants.MAX_GENOTYPE_QUAL} //check that QG is capped at
```
Comment (akiezun) [inline src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java]: text got cut off

**Comment by akiezun** [comment]
looks good. 2 small edits and merge away

File: src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java (hunk 1 above)
Comment (lbergelson) [inline src/main/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlock.java]: fixed

File: src/test/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriterUnitTest.java (hunk 2 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/variant/writers/GVCFWriterUnitTest.java]: woops.  forgot to switch to the temp file after testing with a file I could find easily. fixed

File: src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java (hunk 3 above)
Comment (lbergelson) [inline src/test/java/org/broadinstitute/hellbender/utils/variant/writers/HomRefBlockUnitTest.java]: fixed

**Pull request merged**